app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookFragmentTest.kt

```diff
@@ -67,10 +74,49 @@ class RoadBookFragmentTest {
             click(),
             RecyclerViewActions.scrollToLastPosition<RoadBookViewAdapter.RecordViewHolder>(),
         )
-        onView((withText(DestinationRecords.RECORD_TO_ADD.destName)))
+        // Record just added is displayed at the end of the list
+        onView((withText(DestinationRecords.RECORD_TO_ADD.destID)))
             .check(matches(isDisplayed()))
         onView(withId(R.id.fab_delete)).perform(click())
-        onView((withText(DestinationRecords.RECORD_TO_ADD.destName)))
+
+        // Record added previously is now deleted
+        onView((withText(DestinationRecords.RECORD_TO_ADD.destID)))
             .check(doesNotExist())
     }
+
+    private val db = Firebase.database.reference
+
+    @Test
+    fun roadBookIsBackedUpCorrectly() {
+        //Navigate to an other Fragment
+        val date = Calendar.getInstance().time
+        val ref = db
+            .child("Sheet-shift")
+            .child(SimpleDateFormat.getDateInstance().format(date))
+
+        onView(withId(R.id.fab)).perform(click()) // Add 1 record
+
+        // Navigate out of the RoadBookFragment
+        onView(withId(R.id.drawer_layout))
+            .perform(DrawerActions.open())
+        onView(withId(R.id.directoryFragment))
+            .perform(click())
+        onView(withId(R.id.fragment_directory_directors_parent))
+            .check(matches(isDisplayed()))
+
+        val future = CompletableFuture<List<String>>()
+        ref.get().addOnSuccessListener {
+            if (it.value == null) {
+                future.completeExceptionally(NoSuchFieldException())
+            }
```

Comment:
Even though we're in a testing class it can always help to add a few comments to understand perfectly what the tests are doing.